lib/admin/components/ProjectAccessSettings.js

```diff
@@ -57,27 +58,16 @@ export default class ProjectAccessSettings extends Component<Props, State> {
     this.props.projectAccessUpdated(this.props.project.id, access)
   }
 
-  feedsUpdated = () => {
+  feedsUpdated = (evt: SyntheticInputEvent<HTMLInputElement>) => {
     const {project, projectFeedsUpdated} = this.props
-    const selectedFeeds = []
-    if (project.feedSources) {
-      project.feedSources.forEach((feed) => {
-        // $FlowFixMe
-        var checkbox = this[`feed-${feed.id}`]
-        if (checkbox.checked) selectedFeeds.push(feed.id)
-      })
-    }
-    projectFeedsUpdated(project.id, selectedFeeds)
+    projectFeedsUpdated(project.id, evt.target.name, evt.target.checked)
   }
 
-  permissionsUpdated = () => {
-    const selectedPermissions = []
-    allPermissions.forEach((permission) => {
-      // $FlowFixMe
-      var checkbox = this[`permission-${permission.type}`]
-      if (checkbox.checked) selectedPermissions.push(permission.type)
-    })
-    this.props.projectPermissionsUpdated(this.props.project.id, selectedPermissions)
+  permissionsUpdated = (evt: SyntheticInputEvent<HTMLInputElement>) => {
```

Comment:
Are you OK with me just fixing the name of this handler method instead of adding JS doc? It should be `onTogglePermission` -- that may be where some of the confusion is coming from.